lib/veil/credential_collection/chef_secrets_env.rb

```diff
@@ -0,0 +1,38 @@
+require "veil/credential_collection/base"
+require "json"
+
+module Veil
+  class CredentialCollection
+    class ChefSecretsEnv < Base
+
+      # Create a new ChefSecretsEnv
+      #
+      # @param [Hash] opts
+      #   a hash of options to pass to the constructor
+      def initialize(opts = {})
+        var_name = opts[:var_name] || 'CHEF_SECRETS_DATA'
+
+        @credentials = {}
+        import_credentials_hash(read_env_var(var_name))
+      end
+
+      # Unsupported methods
+      def rotate
+        raise NotImplementedError
+      end
+      alias_method :rotate_credentials, :rotate
+      alias_method :save, :rotate
+
+      def read_env_var(var_name)
+        value = ENV[var_name]
+        unless value
+          raise InvalidCredentialCollectionEnv.new("Env var #{var_name} does not exist")
+        end
+
+        JSON.parse(value)
```

Comment:
Could we wrap this in a rescue and  raise `InvalidCredentialsCollectionEnv` with the message out of the JSON error? Callers shouldn't have to know that we could raise a JSON parsing error - that's an internal detail.